ratelimit/record_handlers/helpers.py

```diff
@@ -0,0 +1,123 @@
+from datetime import timedelta
+from hashlib import md5
+from typing import Callable, Optional
+
+import ipware.ip
+from django.core.cache import BaseCache, caches
+from django.utils.module_loading import import_string
+from ratelimit.conf import settings
+
+
+def get_client_ip_address(request) -> str:
```

Comment:
There's already a tool that does that so that we don't need to install an external package (i.e. ipware): https://github.com/jsocol/django-ratelimit/blob/651c73074e43e6e501871baad0f1c2eb5cbbbe13/django_ratelimit/core.py#L29-L59